common/gpu/nvidia/ada-lovelace/default.nix

```diff
@@ -7,4 +7,22 @@ in
 
   # enable the open source drivers if the package supports it
   hardware.nvidia.open = lib.mkOverride 990 (nvidiaPackage ? open && nvidiaPackage ? firmware);
+
+  # nvidia's hibernate, suspend, and resume services are not normally triggered on suspend-then-hibernate and hybrid-hibernate
```

Comment:
Is this something the service should do in general or just for this service?

Reply:
So that has to do with nvidia's services to suspend and resume the graphics card. If you read the service file from the driver, you will notice it triggers on the normal suspend/hibernate/resume systemd services, but not the "special" ones such as suspend-then-hibernate or hybrid-sleep. The normal options will work, but this causes the card to fail in the special modes with:
PM: pci_pm_suspend(): nv_pmops_suspend+0x0/0x30 [nvidia] returns -5
PM: dpm_run_callback(): pci_pm_suspend+0x0/0x170 returns -5
nvidia 0000:07:00.0: PM: failed to suspend: error -5
PM: Some devices failed to suspend, or early wake event detected

See here for issue discussion: https://forums.developer.nvidia.com/t/nvidia-laptop-unable-to-suspend/221788/10 
See here for solution: https://forums.developer.nvidia.com/t/systemds-suspend-then-hibernate-not-working-in-nvidia-optimus-laptop/213690

I was hesitant to add it to the common/gpu/nvidia/default.nix since I did not want to break something I was unaware of. I believe I have read it is not an issue on older nvidia GPUs. However, I know this is an issue with the nvidia services, not just the Zephyrus and have had similar problems on my Surface Book 3 (1660 Ti Max-Q). Perhaps it should live in common/gpu/nvidia/default.nix and check the driver version? Or should I make it an option with a default of false?

Reply:
Maybe we can gradually roll this out until we can see a pattern emerge.